samples/DotNet/Microsoft.Azure.ServiceBus/SessionSendReceiveUsingSessionClient/Program.cs

```diff
@@ -0,0 +1,107 @@
+        const string QueueName = "{Queue Name of a Queue that supports sessions}";
+        static IMessageSender messageSender;
+        static ISessionClient sessionClient;
+
+        static void Main(string[] args)
+        {
+            MainAsync().GetAwaiter().GetResult();
+        }
+
+        static async Task MainAsync()
+        {
+            const int numberOfSessions = 5;
+            const int numberOfMessagesPerSession = 3;
+
+            messageSender = new MessageSender(ServiceBusConnectionString, QueueName);
+            sessionClient = new SessionClient(ServiceBusConnectionString, QueueName);
+
+            // Send messages with sessionId set
+            await SendSessionMessagesAsync(numberOfSessions, numberOfMessagesPerSession);
+
+            // Receive all Session based messages using SessionClient
+            await ReceiveSessionMessagesAsync(numberOfSessions, numberOfMessagesPerSession);
+
+            Console.WriteLine("=========================================================");
+            Console.WriteLine("Completed Receiving all messages... Press any key to exit");
+            Console.WriteLine("=========================================================");
+
+            Console.ReadKey();
+
+            await messageSender.CloseAsync();
+            await sessionClient.CloseAsync();
+        }
+
+        static async Task ReceiveSessionMessagesAsync(int numberOfSessions, int messagesPerSession)
+        {
+            while(numberOfSessions-- > 0)
+            {
+                int messagesReceivedPerSession = 0;
+                IMessageSession session = await sessionClient.AcceptMessageSessionAsync();
+                if(session != null)
+                {
+                    while(messagesReceivedPerSession++ < messagesPerSession)
+                    {
+                        Message message = await session.ReceiveAsync();
+
+                        Console.WriteLine($"Received Session: {session.SessionId} message: SequenceNumber: {message.SystemProperties.SequenceNumber}");
+
+                        // Complete the message so that it is not received again.
+                        // This can be done only if the queueClient is created in ReceiveMode.PeekLock mode (which is default).
+                        await session.CompleteAsync(message.SystemProperties.LockToken);
+                    }
+
+                    // Close the Session after receiving all messages from the session
+                    await session.CloseAsync();
+                }
+            }
+        }
+
+        static async Task SendSessionMessagesAsync(int numberOfSessions, int messagesPerSession)
+        {
+            const string SessionPrefix = "session";
+
+            if (numberOfSessions == 0 || messagesPerSession == 0)
+            {
+                await Task.FromResult(false);
+            }
+
+            for (int i = 0; i < numberOfSessions; i++)
```

Comment:
We should showcase the FIFO ordering while receiving. So while sending, sessionId could be randomly picked up from a list of prebuilt sessionIDs and sent, and while receiving they will all be processed in FIFO ordering for session